src/main/java/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package service.operation;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+
+public class PurchaseOperation implements OperationHandler {
+    private FruitDao fruitDao;
+
+    public PurchaseOperation() {
+        this.fruitDao = new FruitDaoImpl();
+    }
+
+    @Override
+    public void handle(String fruit, int quantity) {
+        int balanceAfterPurchase = fruitDao.getFruitBalance(fruit) - quantity;
+        if (balanceAfterPurchase >= 0) {
+            fruitDao.updateBalanceOfFruit(fruit,balanceAfterPurchase);
+        } else {
+            throw new RuntimeException("We don't have enough " + fruit + "s");
+        }
```

Comment:
Handling the Purchase operation requires checking the result balance before saving it in the Storage - it should be positive. The current implementation correctly throws a `RuntimeException` if the balance is negative, which is good. However, consider adding more details to the error message, such as the attempted purchase quantity and the available balance, to provide more context for the error.